src/Microsoft.TestPlatform.Common/ExtensionFramework/Utilities/TestExtensions.cs

```diff
@@ -390,4 +485,42 @@ internal void InvalidateCache()
         }
     }
 
+    private void AddCachedExtensionToExtensionMap<T>(
+        Dictionary<string, HashSet<string>> extensionMap,
+        string extensionType,
+        IList<T> extensions) where T : TestPluginInformation
```

Comment:
This could be only `IEnumerable<T>` as you only check for `null` and then `Select`.

Reply:
Thanks for pointing it out, fixed.